src/object/srv_obj.c

```diff
@@ -357,6 +357,19 @@ obj_bulk_transfer(crt_rpc_t *rpc, crt_bulk_op_t bulk_op, bool bulk_bind,
 		rc = ret ? dss_abterr2der(ret) : *status;
 
 	ABT_eventual_free(&arg.eventual);
+	/* After RDMA is done, corrupt the server data */
+	if (DAOS_FAIL_CHECK(DAOS_CHECKSUM_SDATA_CORRUPT)) {
```

Comment:
should this not check whether checksum is enabled on the container?

Reply:
Yes, Presently, it doesn't check. Without checksum enabled, DAOS will ignore the corruption.
Do you want the change in this PR or can I create a separate PR and work on those changes?